File: PTJ4-Shared-Class-Port/starter/java/BankAccount.java
public class BankAccount {
	private final String owner;
	private double balance;

	public BankAccount(String owner, double balance) {
		this.owner = owner;
		this.balance = balance;
	}

	public void deposit(double amount) {
		// TODO: add amount to balance
	}

	public boolean withdraw(double amount) {
		// TODO: reject withdrawals larger than balance
		return false;
	}

	public String summary() {
		// TODO: return a compact summary string
		return "";
	}
}

File: PTJ4-Shared-Class-Port/starter/cpp/BankAccount.cpp
#include "BankAccount.h"

BankAccount::BankAccount(std::string owner, double balance)
	: owner_(std::move(owner)), balance_(balance) {}

void BankAccount::deposit(double amount) {
	// TODO: add amount to the balance
}

bool BankAccount::withdraw(double amount) {
	// TODO: reject withdrawals larger than the balance
	return false;
}

std::string BankAccount::summary() const {
	// TODO: return a compact summary string
	return "";
}
